src/client/signal.ml

```diff
@@ -0,0 +1,40 @@
+module Trace_service = Opentelemetry.Proto.Trace_service
+module Metrics_service = Opentelemetry.Proto.Metrics_service
+module Logs_service = Opentelemetry.Proto.Logs_service
+module Span = Opentelemetry.Span
+
+let ( let@ ) f x = f x
+
+module Converter () = struct
```

Comment:
ok, more general note. I would prefer to remove the functor and have an explicit `state` type passed around (that's basically just `type state = { encode: Pbrt.Encoder.t } [@@unboxed]` for now). This way it's easier to embed in whatever thread-local/domain-local/state-machine needs it.

Reply:
This has been fixed in the way we discussed. Thanks!